### PR TITLE
refactor: Linter issues and general refactor

### PR DESCRIPTION
**Pull request by facundomedica**

### CODE REVIEW

File: cmd/peggo/client/cosmos.go [hunk 1]
```diff
@@ -408,32 +405,32 @@ func (c *cosmosClient) runBatchBroadcast() {
...
 				c.txFactory = c.txFactory.WithSequence(c.accSeq)
 				c.txFactory = c.txFactory.WithAccountNumber(c.accNum)
-				log.Debugln("retrying broadcastTx with nonce", c.accSeq)
+				c.logger.Debug().Uint64("nonce", c.accSeq).Msg("retrying broadcastTx with nonce")
 				res, err = c.broadcastTx(c.ctx, c.txFactory, true, toSubmit...)
 			}
 			if err != nil {
 				resJSON, _ := json.MarshalIndent(res, "", "\t")
-				c.logger.WithField("size", len(toSubmit)).WithError(err).Errorln("failed to commit msg batch:", string(resJSON))
+				c.logger.Err(err).Int("size", len(toSubmit)).RawJSON("resJSON", resJSON).Msg("failed to commit msg batch")
```
Comment (alexanderbez) [inline cmd/peggo/client/cosmos.go:420]:
```suggestion
				c.logger.Err(err).Int("size", len(toSubmit)).RawJSON("tx_response", resJSON).Msg("failed to (sync) broadcast batch tx")
```

File: cmd/peggo/client/cosmos.go [hunk 2]
```diff
@@ -408,32 +405,32 @@ func (c *cosmosClient) runBatchBroadcast() {
...
 				return
 			}
 		}
 
 		if res.Code != 0 {
 			err = errors.Errorf("error %d (%s): %s", res.Code, res.Codespace, res.RawLog)
-			log.WithField("txHash", res.TxHash).WithError(err).Errorln("failed to commit msg batch")
+			c.logger.Err(err).Str("txHash", res.TxHash).Msg("failed to commit msg batch")
```
Comment (alexanderbez) [inline cmd/peggo/client/cosmos.go:427]:
```suggestion
			c.logger.Err(err).Str("txHash", res.TxHash).Msg("failed to (sync) broadcast batch tx")
```

File: cmd/peggo/client/cosmos.go [hunk 3]
```diff
@@ -408,32 +405,32 @@ func (c *cosmosClient) runBatchBroadcast() {
...
 		} else {
-			log.WithField("txHash", res.TxHash).Debugln("msg batch committed successfully")
+			c.logger.Debug().Str("txHash", res.TxHash).Msg("msg batch committed successfully")
```
Comment (alexanderbez) [inline cmd/peggo/client/cosmos.go:429]:
```suggestion
			c.logger.Debug().Str("txHash", res.TxHash).Msg("batch tx committed successfully")
```

File: cmd/peggo/flags.go [hunk 4]
```diff
@@ -58,7 +58,11 @@ func cosmosKeyringFlagSet() *pflag.FlagSet {
 	fs.String(flagCosmosKeyring, keyring.BackendFile, "Specify Cosmos keyring backend (os|file|kwallet|pass|test)")
 	fs.String(flagCosmosKeyringDir, "", "Specify Cosmos keyring directory, if using file keyring")
 	fs.String(flagCosmosKeyringApp, "peggo", "Specify Cosmos keyring app name")
-	fs.String(flagCosmosFrom, "", "Specify the Cosmos validator key name or address. If specified, must exist in keyring, ledger or match the privkey")
+	fs.String(
```
Comment (alexanderbez) [inline cmd/peggo/flags.go:61]: I think we can `// nolint: lll` for files with flags in them 👍

File: orchestrator/eth_event_watcher.go [hunk 5]
```diff
@@ -148,11 +151,13 @@ func (p *peggyOrchestrator) CheckForEvents(
...
-		"start":     startingBlock,
-		"end":       currentBlock,
-		"Withdraws": transactionBatchExecutedEvents,
-	}).Debugln("Scanned TransactionBatchExecuted events from Ethereum")
+
+	p.logger.Debug().
+		Uint64("start", startingBlock).
+		Uint64("end", currentBlock).
+		// TODO: fix log. print the list of events? Or the length of it?
+		Int("events", len(transactionBatchExecutedEvents)).
```
Comment (alexanderbez) [inline orchestrator/eth_event_watcher.go:159]:
```suggestion
		Int("num_events", len(transactionBatchExecutedEvents)).
```

File: orchestrator/eth_event_watcher.go [hunk 6]
```diff
@@ -116,11 +118,12 @@ func (p *peggyOrchestrator) CheckForEvents(
...
-	log.WithFields(log.Fields{
-		"start":    startingBlock,
-		"end":      currentBlock,
-		"Deposits": sendToCosmosEvents,
-	}).Debugln("Scanned SendToCosmos events from Ethereum")
+	p.logger.Debug().
+		Uint64("start", startingBlock).
+		Uint64("end", currentBlock).
+		// TODO: fix log. print the list of events? Or the length of it?
+		Int("events", len(sendToCosmosEvents)).
```
Comment (alexanderbez) [inline orchestrator/eth_event_watcher.go:125]:
```suggestion
		Int("num_events", len(sendToCosmosEvents)).
```

File: orchestrator/eth_event_watcher.go [hunk 7]
```diff
@@ -82,11 +81,14 @@ func (p *peggyOrchestrator) CheckForEvents(
...
-		"end":       currentBlock,
-		"Contracts": erc20DeployedEvents,
-	}).Debugln("Scanned ERC20Deployed events from Ethereum")
+	p.logger.Debug().
+		Uint64("start", startingBlock).
+		Uint64("end", currentBlock).
+		// TODO: fix log. print the list of events? Or the length of it?
+		// we are only seeing the pointer address of the objects in the slice
+		// Object("erc20DeployedEvents", erc20DeployedEvents)).
+		Int("events", len(erc20DeployedEvents)).
```
Comment (alexanderbez) [inline orchestrator/eth_event_watcher.go:90]:
```suggestion
		Int("num_events", len(erc20DeployedEvents)).
```

File: orchestrator/eth_event_watcher.go [hunk 8]
```diff
@@ -82,11 +81,14 @@ func (p *peggyOrchestrator) CheckForEvents(
...
 
-	log.WithFields(log.Fields{
-		"start":     startingBlock,
-		"end":       currentBlock,
-		"Contracts": erc20DeployedEvents,
-	}).Debugln("Scanned ERC20Deployed events from Ethereum")
+	p.logger.Debug().
+		Uint64("start", startingBlock).
+		Uint64("end", currentBlock).
+		// TODO: fix log. print the list of events? Or the length of it?
```
Comment (alexanderbez) [inline orchestrator/eth_event_watcher.go:87]: Lets remove the TODO or create an issue for it and reference it.

File: orchestrator/cosmos/broadcast.go [hunk 9]
```diff
@@ -361,14 +364,14 @@ func (s *peggyBroadcastClient) sendERC20DeployedClaims(
...
 	}
 
-	log.WithFields(log.Fields{
-		"event_nonce": event.EventNonce.String(),
-		"txHash":      txResponse.TxHash,
-	}).Infoln("Oracle sent ERC20Deployed event succesfully")
+	s.logger.Info().
+		Str("txHash", txResponse.TxHash).
+		Str("eventNonce", event.EventNonce.String()).
+		Msg("Oracle sent ERC20Deployed event successfully")
```
Comment (alexanderbez) [inline orchestrator/cosmos/broadcast.go:362]:
```suggestion
		Msg("oracle sent ERC20Deployed event successfully")
```

File: orchestrator/cosmos/broadcast.go [hunk 10]
```diff
@@ -342,11 +345,11 @@ func (s *peggyBroadcastClient) sendERC20DeployedClaims(
...
-	log.WithFields(log.Fields{
-		"token_contract": event.TokenContract.Hex(),
-		"cosmos_denom":   event.CosmosDenom,
-		"event_nonce":    event.EventNonce.String(),
-	}).Infoln("Oracle observed an ERC20 deployed event. Sending MsgERC20DeployedClaim")
+	s.logger.Info().
+		Str("eventNonce", event.EventNonce.String()).
+		Str("tokenContract", event.TokenContract.Hex()).
+		Str("cosmosDenom", event.CosmosDenom).
+		Msg("Oracle observed an ERC20 deployed event. Sending MsgERC20DeployedClaim")
```
Comment (alexanderbez) [inline orchestrator/cosmos/broadcast.go:340]:
```suggestion
		Msg("oracle observed an ERC20 deployed event. Sending MsgERC20DeployedClaim")
```

File: orchestrator/cosmos/broadcast.go [hunk 11]
```diff
@@ -325,14 +328,14 @@ func (s *peggyBroadcastClient) sendValsetUpdateClaims(
...
 	}
 
-	log.WithFields(log.Fields{
-		"event_nonce": valsetUpdate.EventNonce.String(),
-		"txHash":      txResponse.TxHash,
-	}).Infoln("Oracle sent ValsetUpdate event succesfully")
+	s.logger.Info().
+		Str("txHash", txResponse.TxHash).
+		Str("eventNonce", valsetUpdate.EventNonce.String()).
+		Msg("Oracle sent ValsetUpdate event successfully")
```
Comment (alexanderbez) [inline orchestrator/cosmos/broadcast.go:329]:
```suggestion
		Msg("oracle sent ValsetUpdate event successfully")
```

File: orchestrator/cosmos/broadcast.go [hunk 12]
```diff
@@ -296,14 +299,14 @@ func (s *peggyBroadcastClient) sendValsetUpdateClaims(
...
-		"rewardToken":  valsetUpdate.RewardToken.Hex(),
-	}).Infoln("Oracle observed a valsetUpdate event. Sending MsgValsetUpdatedClaim")
+	s.logger.Info().
+		Str("eventNonce", valsetUpdate.EventNonce.String()).
+		Uint64("valsetNonce", valsetUpdate.NewValsetNonce.Uint64()).
+		Int("validators", len(valsetUpdate.Validators)).
+		Interface("powers", valsetUpdate.Powers).
+		Uint64("rewardAmount", valsetUpdate.RewardAmount.Uint64()).
+		Str("rewardToken", valsetUpdate.RewardToken.Hex()).
+		Msg("Oracle observed a valset update event. Sending MsgValsetUpdateClaim")
```
Comment (alexanderbez) [inline orchestrator/cosmos/broadcast.go:300]:
```suggestion
		Msg("oracle observed a valset update event. Sending MsgValsetUpdateClaim")
```

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Nice work @facundomedica! But I'd like to strive for consistency. Can we make sure we address all TODOs and ensure consistency in log fields (snake-case) and no capitalization in logs?

File: orchestrator/eth_event_watcher.go (hunk 8 above)
Comment (facundomedica) [reply orchestrator/eth_event_watcher.go:87]: We've already discussed this. Gonna print length of lists

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**